src/main/scala/com/stackstate/pac4j/store/SessionStorage.scala

```diff
@@ -0,0 +1,47 @@
+package com.stackstate.pac4j.store
+
+import scala.concurrent.duration.FiniteDuration
+
+object SessionStorage {
+  type SessionKey = String
+  type ValueKey = String
+}
+
+trait SessionStorage {
+  import SessionStorage._
+
+  /**
+    * Time after which a session gets automatically destroyed
+    */
+  val sessionLifetime: FiniteDuration
+
+  /**
+    * Ensure existance of a session with the provided key. Returns whether the session is new
+    */
+  def ensureSession(key: SessionKey): Boolean
```

Comment:
maybe rename this method to `createIfNeeded` or something like that.

Reply:
done